main_test.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"os/exec"
+)
+
+func init() {
+	// import the github.com SSH public key
+	cmd := exec.Command("sh", "-c", "! [ -f $HOME/.ssh/known_hosts ] && "+
+		"mkdir -p $HOME/.ssh && ssh-keyscan github.com | "+
+		"tee -a $HOME/.ssh/known_hosts && chmod 600 $HOME/.ssh/known_hosts || true")
+	err := cmd.Run()
+	if err != nil {
+		panic(err)
+	}
+	// import the SSH private key
+	cmd = exec.Command("sh", "-c", "[ -n \"$SSH_PRIVATE_KEY\" ] && ! [ -f $HOME/.ssh/id_rsa ] && "+
+		"echo \"$SSH_PRIVATE_KEY\" | tr -d '\r' >> $HOME/.ssh/id_rsa && chmod 600 $HOME/.ssh/id_rsa || true")
+	err = cmd.Run()
```

Comment:
I don't understand what is the intention here. We are not going to set an SSH_PRIVATE_KEY in CI variables, so is only meant for local use? Can the tests run without the SSH key at all?

Reply:
Without the SSH key, the unit tests cannot clone the repos, thus unit tests won't run in CI. Am I wrong?

Reply:
Exactly, but we cannot have an *unprotected* SSH key on CI variables. I saw we have it right now and I assumed you added it while developing, but we definitely have to remove before merge, so the tests won't be able to run.

Reply:
ok, which options do I have, then? maybe we can discuss it on slack.

Reply:
reworked to work with https links when running unit tests